smdebug/pytorch/hook.py

```diff
@@ -118,6 +129,47 @@ def _prepare_collections(self):
 
     # This hook is invoked by trainer prior to running the forward pass.
     def forward_pre_hook(self, module, inputs):
+        if self.profiler_enabled and self.forward_start != 0:
+            self.timeline_writer.write_trace_events(
+               op_name="forward",
+               #timestamp=time.time(),
+               duration=self.forward_end-self.forward_start,
+               step_num=self.step)
+            self.timeline_writer.write_trace_events(
+               op_name="backward",
+               #timestamp=time.time(),
+               duration=self.backward_end-self.backward_start,
+               step_num=self.step)
+            self.timeline_writer.flush()
+            if self.step in self.profiler_steps:
+                if not self.autograd_profiler_enabled:
+                   torch.autograd._enable_profiler(torch.autograd.ProfilerConfig(torch.autograd.ProfilerState.CUDA, False))
+                   self.autograd_profiler_enabled = True
+                   self.start_profiler = time.time() * 1000000
```

Comment:
you think we should save this 1000000 as a constant that can be used throughout smdebug?

Reply:
Yes, that's a good idea. Shall we add it to smdebug/core/config_constants.py ?

Reply:
yes, that would be good. I'll update my PR with same changes once you make it

Reply:
added this to config constant in PR 247.